zsh-syntax-highlighting.zsh

```diff
@@ -42,6 +42,13 @@ if true; then
   fi
 fi
 
+integer zsh_highlight_use_redrawhook
+if autoload -U +X -- add-zle-hook-widget 2>/dev/null;
+   [[ "${${(@f)"$(which -- add-zle-hook-widget)"}[2]}" != $'\t'$histchars[3]' undefined' ]];
+then
+  (( zsh_highlight_use_redrawhook=1 ))
+fi
+
```

Comment:
Do we need this section? Instead in ---- SETUP ---- we could simply do:

if "add-zle-hook-widget is availabe"; then
  "add-zle-hook-widget"
else
  "bind widgets"
fi

That would ready clearer, imho.